packages/astro/src/core/app/node.ts

```diff
@@ -116,11 +117,27 @@ export class NodeApp extends App {
 		}
 		return super.match(req, opts);
 	}
-	render(req: NodeIncomingMessage | Request, routeData?: RouteData, locals?: object) {
+	render(request: NodeIncomingMessage | Request, options?: RenderOptions): Promise<Response>
+	render(request: NodeIncomingMessage | Request, routeData?: RouteData, locals?: object): Promise<Response>
+	render(req: NodeIncomingMessage | Request, routeDataOrOptions?: RouteData | RenderOptions, locals?: object) {
+		let routeData: RouteData | undefined;
+		
+		if (routeDataOrOptions && ('routeData' in routeDataOrOptions || 'locals' in routeDataOrOptions)) {
+			if ('routeData' in routeDataOrOptions) {
+				routeData = routeDataOrOptions.routeData;
+			}
+			if ('locals' in routeDataOrOptions) {
+				locals = routeDataOrOptions.locals;
+			}
+		}
+		else {
+			routeData = routeDataOrOptions as RouteData;
```

Comment:
Should we log a deprecation warning here? Maintainers probably won't catch this change unless we have visible log.

Reply:
![image](https://github.com/withastro/astro/assets/69170106/4d24f1dc-f50c-4aa2-8851-636c58a4a65c)